Description: Hide error from dlsym()
 dlsym(), starting in glibc 2.24 actually reports errors. In our case,
 we try to get ACL functions which are not in the glibc. This causes
 failures in test suites, so hide those messages for non-debugging
 purposes for now. It also makes the build logs annoying to read.
Author: Julian Andres Klode <juliank@ubuntu.com>
Origin: vendor
Bug-Debian: https://bugs.debian.org/830912
Forwarded: no
Last-Update: 2016-08-12

--- a/libfakeroot.c
+++ b/libfakeroot.c
@@ -256,10 +256,16 @@ void load_library_symbols(void){
  /* clear dlerror() just in case dlsym() legitimately returns NULL */
     msg = dlerror();
     *(next_wrap[i].doit)=dlsym(get_libc(), next_wrap[i].name);
+
     if ( (msg = dlerror()) != NULL){
-      fprintf (stderr, "dlsym(%s): %s\n", next_wrap[i].name, msg);
-/*    abort ();*/
+#ifdef LIBFAKEROOT_DEBUGGING
+      if (fakeroot_debug) {
+        fprintf (stderr, "dlsym(%s): %s\n", next_wrap[i].name, msg);
+/*      abort ();*/
+      }
+#endif
     }
+
   }
 }
 
